pubtools/pulplib/_impl/client/search.py

```diff
@@ -113,3 +117,34 @@ def field_match(to_match):
         return {"$lt": to_mongo_json(to_match._value)}
 
     raise TypeError("Not a matcher: %s" % repr(to_match))
+
+
+def validate_type_ids(type_ids):
+    valid_type_ids = []
+    invalid_type_ids = []
+
+    if isinstance(type_ids, str):
+        type_ids = [type_ids]
+
+    if not isinstance(type_ids, (list, tuple)):
```

Comment:
minor, but I think you basically what to know if hasattr(type_ids, "\_\_iter\_\_")

Reply:
Not sure that we want or need to open this up to other iterators, though.